lib.php

```diff
@@ -883,8 +885,7 @@ function moodleoverflow_send_mails() {
 
                 // Preapare to actually send the post now. Build up the content.
                 $cleanname = str_replace('"', "'", strip_tags(format_string($moodleoverflow->name)));
-                $coursecontext = context_course::instance($course->id);
-                $shortname = format_string($course->shortname, true, ['context' => $coursecontext]);
+                $shortname = format_string($course->shortname, true, ['context' => context_course::instance($course->id)]);
```

Comment:
Here and previously you still pass an object to the format_string function. 
> The $options array must not be a \context or \core\context object. This was never supported and behaviour was previously undetected

I guess you are supposed to just pass the name, but please check! (Or correct me if I misunderstood it!)

Reply:
As I understand it (but I'm a little bit confused) the array itself should not be an context-object (like function(param1, param2, object) ) but inside the array an object is okay ( like function(param1, param2, array('key' => contextobject)) )

Reply:
Hm, I looked again at it and I think it is illegal to do sth like this: 
`format_string($course->shortname, true, context_course::instance($course->id));`
(see function) [https://github.com/moodle/moodle/blob/757be30c3903bed55ad4a3607346b23510741745/lib/weblib.php#L1468]
Therefore, I guess the previous way was also okay just passing without making it an array is illegal.

Reply:
okay, im changing it back as it was